workitem_blackbox_test.go

```diff
@@ -130,3 +137,162 @@ func TestListByFields(t *testing.T) {
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Malformed Bearer %s", tokenStr)
+}
+
+func getValidAuthHeader(t *testing.T, key interface{}) string {
+	token := jwt.New(jwt.SigningMethodRS256)
+	tokenStr, err := token.SignedString(key)
+	if err != nil {
+		t.Fatal("Could not sign the token ", err)
+	}
+	return fmt.Sprintf("Bearer %s", tokenStr)
+}
+
+func TestUnauthorizeCreateWI(t *testing.T) {
+	resource.Require(t, resource.Database)
+
+	privatekey, err := jwt.ParseRSAPrivateKeyFromPEM(([]byte(RSAPrivateKeyTest)))
+	if err != nil {
+		t.Fatal("Could not parse Key ", err)
+	}
+	publickey, err := jwt.ParseRSAPublicKeyFromPEM(([]byte(RSAPublicKeyTest)))
+	if err != nil {
+		t.Fatal("Could not parse Key ", err)
+	}
+
+	differentPrivatekey, err := jwt.ParseRSAPrivateKeyFromPEM(([]byte(RSADifferentPrivateKeyTest)))
+	// Build Token set to test
+	// following structure represents key as a token string and int as a expected status code
+	tokenTests := map[string]int{
+		getExpiredAuthHeader(t, privatekey):        401,
+		getMalformedAuthHeader(t, privatekey):      401,
+		getValidAuthHeader(t, differentPrivatekey): 401,
+		"": 401,
+	}
+
+	for authToken, _ := range tokenTests {
+		payload := []byte(`{"type": "system.userstory" ,"fields": { "system.creator": "tmaeder", "system.state": "new", "system.title": "My special story", "system.description": "description" }}`)
+
+		// Build a request
+		req, err := http.NewRequest("POST", "/api/workitems", bytes.NewBuffer(payload))
+		if err != nil {
+			t.Fatal("could not create a HTTP request")
+		}
+		// Add Authorization Header
+		req.Header.Add("Authorization", authToken)
+
+		rr := httptest.NewRecorder()
+		ts := models.NewGormTransactionSupport(DB)
+		wir := models.NewWorkItemTypeRepository(ts)
+		repo := models.NewWorkItemRepository(ts, wir)
+
+		// temperory service for testing the middleware
+		service := goa.New("TestUnauthorizedCreateWI-Service")
+		assert.NotNil(t, service)
+
+		// append a middleware to service. Use appropriate RSA keys
+		jwtMiddleware := goajwt.New(publickey, nil, app.NewJWTSecurity())
+		app.UseJWTMiddleware(service, jwtMiddleware)
+
+		controller := NewWorkitemController(service, repo, ts)
+		app.MountWorkitemController(service, controller)
+
+		// Hit the service with own request
+		service.Mux.ServeHTTP(rr, req)
+		// the problem I am facing right now is
+		// When no token is set in Authorization header, it throws -- uncaught error err=[NNCptvpI] 401 jwt_security_error: missing header "Authorization"
+		// When wrong formatted token is set in header, it throws -- 401 jwt_security_error: invalid or malformed
+		// This is as expected till now,
+		// But problem = it `throws error` and prints out as `uncaught_error`
+		// Hence I can not capture response status code. any suggestions ?
+		fmt.Println(rr.Code, rr.Body) // always prints 500 because of uncaught_error
+		assert.NotEqual(t, 201, rr.Code)
```

Comment:
Have you tried adding the ErrorHandler middleware? https://github.com/almighty/almighty-core/blob/master/main.go#L119